oak-segment-tar/src/main/java/org/apache/jackrabbit/oak/segment/Segment.java

```diff
@@ -217,6 +217,21 @@ public String toString() {
         }
     }
 
+    public Segment(
+            @NotNull SegmentReader reader,
+            @NotNull SegmentId id,
+            @NotNull SegmentData data,
+            @NotNull RecordNumbers recordNumbers,
+            @NotNull SegmentReferences segmentReferences
+    ) {
+        this.reader = requireNonNull(reader);
+        this.id = requireNonNull(id);
+        this.data = requireNonNull(data);
+        this.recordNumbers = requireNonNull(recordNumbers);
+        this.segmentReferences = requireNonNull(segmentReferences);
+        this.version = LATEST_VERSION;
```

Comment:
I think the version should be read from SegmentData.
```suggestion
        this.version = SegmentVersion.fromByte(data.getVersion());
```
Or is there a specific reason why LATEST_VERSION must be used here? Can you also please add a test for this constructor. E.g. verifying the version is as expected.

Reply:
I agree, I updated my PR. I also added unit tests for the constructor.